### PR TITLE
add info-windows to maps

### PR DESCRIPTION
**Pull request by oasemota99**
Adds small info window to map markers on main map page. Click on the marker to see the name of the event
fixes basic functionality of #101

### CODE REVIEW

File: capstone/src/main/resources/static/map.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ async function createMap() {
     document.getElementById('map'),
     {center: princetonLatLng, zoom: 16});
 
-   const response = await fetch('get-all-events');
+   const response = await fetch('get-map-events');
```
Comment (lbourret) [inline capstone/src/main/resources/static/map.js:8]: Oh in one of my prs I created a new mapping function for filtered events instead and left all-events as normal bc of the error it was causing on maps. So I think whatever pr gets pushed in first, we can just adjust to those mapping functions?

File: capstone/src/main/resources/static/map.js [hunk 2]
```diff
@@ -40,8 +39,28 @@ function createMarker(event,campusMap) {
   const newMarker = new google.maps.Marker({
     map: campusMap,
     title: event.eventTitle,
-    position: eventPosition
+    position: eventPosition,
   });
+  var eventInfoWindow = createInfoWindow(event, campusMap);
+  var eventContent = '<p id=mapContent>'+ event.eventTitle + '</p>';
```
Comment (meltsufin) [inline capstone/src/main/resources/static/map.js:45]: How can the user save the event from this view? Would it be possible to provide a link to the event details page? I'm not sure we have a separate event details page though, just a popup.

File: capstone/src/main/resources/static/map.js [hunk 3]
```diff
@@ -40,8 +39,28 @@ function createMarker(event,campusMap) {
...
+  newMarker.addListener('click', function() {
+    eventInfoWindow.setContent(eventContent);
```
Comment (meltsufin) [inline capstone/src/main/resources/static/map.js:47]: Would it work to just set the content in the `createInfoWindow()`, and only do the `open()` on the click event?

File: capstone/src/main/resources/static/map.js (hunk 1 above)
Comment (jsheng112) [reply capstone/src/main/resources/static/map.js:8]: So none of you guys use 'get-all-events' anymore? I have a PR that filters out past events for that end point but if you guys are not using it I'll just wait for these functions to be merged.

File: capstone/src/main/resources/static/map.js (hunk 2 above)
Comment (oasemota99) [reply capstone/src/main/resources/static/map.js:45]: There isn't a dedicated page, it's just a popup. I'm not sure if I could make a link to it or not. Also the events on the map are the saved events from the user

File: capstone/src/main/resources/static/map.js (hunk 1 above)
Comment (oasemota99) [reply capstone/src/main/resources/static/map.js]: I think filtering out the past events is really useful, so I'm fine with using that instead

**Pull request merged**